hw/ip/sram_ctrl/doc/_index.md

```diff
@@ -23,27 +26,40 @@ The SRAM controller interfaces with the SRAM data and address scrambling device
 
 ![SRAM Controller Block Diagram](sram_ctrl_blockdiag.svg)
 
-As shown in the blockdiagram above (for `Width = 32`), the SRAM controller is a relatively simple module that only contains a CSR node and a key request interface.
-The actual scrambling mechanism is implemented in the `prim_ram_1p_scr` primitive and has to be instantiated alongside the `sram_ctrl`, together with the appropriately parameterized TL-UL SRAM adapter.
-The reason for this arrangement is that memory blocks (like SRAM and flash) are declared separately from the comportable IPs in the OpenTitan top-level generator tool.
-Also, moving the scrambling mechanism into a primitive makes it possible to conveniently reuse the same scrambling mechanism in other IPs such as [OTBN]({{< relref "hw/ip/otbn/doc" >}}).
+As shown in the block diagram above, the SRAM controller contains a TL-UL adapter, an initialization LFSR, the CSR node, key request logic and an instance of `prim_ram_1p_scr` that implements the actual scrambling mechanism.
 
-The scrambling mechanism is always enabled, but the `sram_ctrl` provides the scrambling device with a predefined scrambling key and nonce when it comes out of reset.
+The SRAM controller supports the system-wide end-to-end bus integrity scheme and thus stores the data integrity bits alongside each data word in the memory.
+I.e., this means that both the 32 data bits and 7 integrity bits are passed through the scrambling device.
+
+Sub-word write operations therefore perform a read-modify-write operation in order to ensure consistency of the integrity bits.
+Hence, the throughput of sub-word write operations is three times lower than for full-word write operations.
+Note however that the throughput of read operations is the same for full- and sub-word read operations.
```

Comment:
do you think it makes sense to add a note here as to why we chose not to do buffered writes? Because I feel like someone will ask.  We could just say it was for design simplicity, and buffered writes for example do not remove the possibility that we still need to stall at some point.

Reply:
yes, let me add that. it's to simplify the design, really...